### PR TITLE
Vendor the download strategy for anu

### PR DESCRIPTION
**Pull request by hmac**
Homebrew have removed the download strategy which supports downloading
assets from a release on a private GitHub repository. We need this to
support anu's distribution mechanism, so we vendor it inside this
repository so anu's formula can use it.

I went down the road of writing tests for this, but I think it's probably more effort than it's worth. Homebrew formulae run inside Homebrew's own environment (and Homebrew's own ruby version). Not only would we need to mock all the class dependencies (`CurlDownloadStrategy`, `GitHub`) but we'd also need to ensure that the `require` calls (to other Homebrew files) work as well. I think we're better off leaving this as-is, and fixing it if it does break in the future. The only users of this tool are other people in the engineering team, so I think this is low risk.

If you want to test this locally:
```bash
cd gocardless/homebrew-taps
git checkout hmac/gh-download
brew uninstall anu
brew tap foo/bar .
brew install anu
brew untap foo/bar
```

You should see that anu installs successfully.

### CODE REVIEW

File: Formula/anu.rb [hunk 1]
```diff
@@ -1,22 +1,28 @@
+# frozen_string_literal: true
```
Comment (benwh) [inline Formula/anu.rb:1]: This is unfortunately going to be removed by GoReleaser, as well as some of the other cosmetic fixes

**Review by benwh — APPROVED** [review]
LGTM - tested and works nicely!

I'm going to leave this PR open, rather than merge straight away, because there's another step required: The `.goreleaser.yml` config in anu needs to be updated. You may want to tweak/reformat the `anu.rb` formula here to match what GoReleaser outputs

File: Formula/anu.rb (hunk 1 above)
Comment (hmac) [reply Formula/anu.rb:1]: That is sad 😢

**Comment by hmac** [comment]
Companion PR: https://github.com/gocardless/anu/pull/1656

I propose merging this first, since it will fix the issue now without requiring a new version of anu to be released. When we do release a new version, it should overwrite the formula harmlessly.

**Pull request merged**